packages/astro/src/vite-plugin-html/transform/escape.ts

```diff
@@ -3,19 +3,21 @@ import type MagicString from 'magic-string';
 import type { Plugin } from 'unified';
 import { visit } from 'unist-util-visit';
 
-import { escape, needsEscape, replaceAttribute } from './utils.js';
+import { escapeTemplateLiteralCharacters, needsEscape, replaceAttribute } from './utils.js';
 
 const rehypeEscape: Plugin<[{ s: MagicString }], Root> = ({ s }) => {
 	return (tree) => {
 		visit(tree, (node: Root | RootContent) => {
 			if (node.type === 'text' || node.type === 'comment') {
 				if (needsEscape(node.value)) {
-					s.overwrite(node.position!.start.offset!, node.position!.end.offset!, escape(node.value));
+					s.overwrite(node.position!.start.offset!, node.position!.end.offset!, escapeTemplateLiteralCharacters(node.value));
 				}
 			} else if (node.type === 'element') {
-				for (const [key, value] of Object.entries(node.properties ?? {})) {
-					const newKey = needsEscape(key) ? escape(key) : key;
-					const newValue = needsEscape(value) ? escape(value) : value;
+				if (!node.properties) return;
+				for (let [key, value] of Object.entries(node.properties)) {
+					key = key.replace(/([A-Z])/g, '-$1').toLowerCase()
```

Comment:
I'm curious about the decision to convert attributes to kebab-case. Could you share the reasoning behind it?
Shouldn't it be left to the browser?

Reply:
Oh this was working around an interesting behavior of one of the rehype parsers. They normalize `kebab-case` attributes into a `camelCase` key on the `node.properties` object, but we need to match the key back to the source code, which was probably using the `kebab-case` format.

This code allows us to match up the parsed AST with the source code for sourcemaps.

Reply:
Ah gotcha, my concern was that an attribute like `MyAttribute` would end up being `-my-attribute`, but that won't happen since the value is always camel case 👍️